Sming/SmingCore/Network/MqttClient.cpp

```diff
@@ -46,22 +63,55 @@ void MqttClient::setKeepAlive(int seconds)
 void MqttClient::setPingRepeatTime(int seconds)
 {
 	if (PingRepeatTime > keepAlive)
-	   PingRepeatTime = keepAlive;
+		PingRepeatTime = keepAlive;
 	else
-	   PingRepeatTime = seconds;
+		PingRepeatTime = seconds;
 }
 
 bool MqttClient::setWill(const String& topic, const String& message, int QoS, bool retained /* = false*/)
 {
 	return mqtt_set_will(&broker, topic.c_str(), message.c_str(), QoS, retained);
 }
 
+bool MqttClient::connect(const URL& url, const String& clientName, uint32_t sslOptions)
+{
+	// eg						"mttqs://frank:frankspassword@frankserver:1883", "unique-client-name");
+	// String Protocol			mqtts
+	// String User;				frank
+	// String Password;			frankspassword
+	// String Host;				frankserver
+	// int Port;				8883  ( usually 1883 for non-ssl )
+	// String Path;				empty
+	// String Query;			empty
+
+	this->server = url.Host;
```

Comment:
Hm... may be it will be best to add in the class a private member called url with type URL and store the port, host, etc there. Which means that the port and server members should be removed completely from the class.